File: front_end/src/entity/Email.php
<?php

declare(strict_types=1);

namespace App\entity;

class Email
{

    public function __construct(
        public readonly int $emailId,
        public readonly String $assunto,
        public readonly String $corpo,
        public readonly String $emailDestinatario,
        public readonly String $emailRemetente,
        public readonly String $status,
        public readonly String $dataEnvio
    ){}


    public static function fromArray(array $data):self
    {
        return new self(
            emailId: $data['emailId'],
            assunto: $data['assunto'],
            corpo: $data['corpo'],
            emailDestinatario: $data['emailDestinatario'],
            emailRemetente: $data['emailRemetente'],
            status: $data['status'],
            dataEnvio: $data['dataEnvio'],
        );
    }
}

/*

"emailId": 9,
			"assunto": "subject",
			"corpo": "body",
			"emailDestinatario": "adm@gmail.com",
			"emailRemetente": "adm@gmail.com",
			"status": "ENVIADO",
			"dataEnvio": "2025-06-01"

*/

File: back_end/api/src/main/java/com/api/api/Model/Email/Email.java
package com.api.api.Model.Email;

import com.api.api.Model.Rascunho.Rascunho;
import com.api.api.Model.User.User;
import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import jdk.jfr.Enabled;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;

import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Table(name = "emails")
@Entity(name = "Email")
@Getter

@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(of = "emailId")
public class Email {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long emailId;

    private String assunto;
    @Lob
    private String corpo;
    private String emailDestinatario;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "email_remetente", referencedColumnName = "email")
    @JsonBackReference
    private User emailRemetente;

    @Enumerated(EnumType.STRING)
    private Status status;

    private LocalDate dataEnvio;


    public Email(CreateData data, User user){
        this.assunto = data.assunto();
        this.corpo = data.corpo();
        this.emailDestinatario = data.emailDestinatario();
        this.emailRemetente = user;

        dataEnvio = LocalDate.now();
        status = Status.ENVIADO;
    }

    public Email(Rascunho rascunho){
        this.assunto = rascunho.getAssunto();
        this.emailDestinatario = rascunho.getEmailDestinatario();
        this.corpo = rascunho.getCorpo();
        this.emailRemetente = rascunho.getEmailRemetente();

        dataEnvio = LocalDate.now();
        status = Status.ENVIADO;
    }



}
